cloud/services/networkinterfaces/networkinterfaces.go

```diff
@@ -151,7 +146,7 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 		})
 
 	if err != nil {
-		return errors.Wrapf(err, "failed to create network interface %s in resource group %s", nicSpec.Name, s.Scope.AzureCluster.Spec.ResourceGroup)
+		return errors.Wrapf(err, "failed to create network interface %s in resource group %s", nicSpec.Name, s.Scope.ResourceGroup())
```

Comment:
We can probably start using fmt.Errorf w/ %w over Wrapf since we're using go 1.13+ now.

Reply:
not required for this PR though

Reply:
yeah I think that can be a refactor everywhere in a separate PR... for now choosing consistency